F50/Preferences/Preferences.ios.cs

```diff
@@ -1,14 +1,142 @@
-﻿using System;
+﻿using Foundation;
+using System;
 using System.Collections.Generic;
 using System.Text;
 
 namespace Xamarin.F50
 {
     public partial class Preferences
     {
-        public void Set(string key, string val)
-        {
+		public bool ContainsKey(string key)
+		{
+			var userDefaults = GetUserDefaults();
+
+			var setting = userDefaults[key];
+			return setting != null;
+		}
 
-        }
-    }
+		public bool Remove(string key)
+		{
+			var userDefaults = GetUserDefaults();
+			try
+			{
+				if (userDefaults[key] != null)
+				{
+					userDefaults.RemoveObject(key);
+					userDefaults.Synchronize();
+				}
+			}
+			catch (Exception ex)
+			{
+				Console.WriteLine("Unable to remove: " + key, " Message: " + ex.Message);
+				return false;
+			}
+
+			return true;
+		}
+
+		public bool Clear()
+		{
+			var userDefaults = GetUserDefaults();
+			try
+			{
+				var items = userDefaults.ToDictionary();
+
+				foreach (var item in items.Keys)
+				{
+					if (item is NSString nsString)
+						userDefaults.RemoveObject(nsString);
+				}
+				userDefaults.Synchronize();
+
+			}
+			catch (Exception ex)
+			{
+				Console.WriteLine("Unable to clear items, Message: " + ex.Message);
+				return false;
+			}
+
+			return true;
+		}
+		
+		bool Set<T>(string key, T value)
+		{
+			var userDefaults = GetUserDefaults();
+			
+			switch (value)
+			{
+				case string s:
+					userDefaults.SetString(s, key);
+					break;
+				case int i:
+					userDefaults.SetInt(i, key);
+					break;
+				case bool b:
+					userDefaults.SetBool(b, key);
+					break;
+				case long l:
+					userDefaults.SetString(l.ToString(), key);

```

Comment:
Should use invariant culture: Convert.ToString(value, System.Globalization.CultureInfo.InvariantCulture) i know i know, but probably best still